### PR TITLE
[DeadCode] Skip indirect variable definition on RemoveNonExistingVarAnnotationRector

### LINKED ISSUE #8253: Incorrect behavior of RemoveNonExistingVarAnnotationRector
# Bug Report

| Subject        | Details              |
| :------------- | :--------------------|
| Rector version | last dev-main      |
| Installed as   | composer dependency  |

## Minimal PHP Code Causing Issue

phpdocs don't require a variable and can be used just before a variable assignment.
the code in the example should be skipped

See https://getrector.com/demo/16ec367a-fdb3-4021-968c-70c9495ad7e9

```php
<?php

final class DemoFile
{
    public function run(bool $param)
    {
        /** @var MyClass<T> */
        $x = new MyClass;
    }

}
```

### Responsible rules

* `RemoveNonExistingVarAnnotationRector`

### PR DESCRIPTION
**Pull request by samsonasik**
Fixes https://github.com/rectorphp/rector/issues/8253

### CODE REVIEW

**Comment by samsonasik** [comment]
Fixed 🎉 /cc @staabm

File: rules-tests/DeadCode/Rector/Node/RemoveNonExistingVarAnnotationRector/Fixture/skip_next_inline_html.php.inc [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+/**
+ * @var string $value
+ */
+
+ ?>
+
+ <div class="<?php echo $value; ?>"></div>
```
Comment (samsonasik) [inline rules-tests/DeadCode/Rector/Node/RemoveNonExistingVarAnnotationRector/Fixture/skip_next_inline_html.php.inc:9]: @staabm this also skip next InlineHTML

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I am merging it ;)

**Comment by staabm** [comment]
thanks

**Pull request merged**